client.go

```diff
@@ -1441,51 +1441,71 @@ type Secret struct {
 }
 
 type CreateSecretOptions struct {
-	SecretType  string
-	SecretName  string
-	SecretValue string
-	Description string
-	FileName    string
-	RealmName   string
+	SecretType       string
+	SecretName       string
+	SecretValue      string
+	Description      string
+	FileName         string
+	RealmName        string
+	OwnerPermissions []string // optional
 }
 
 // CreateSecret makes a secret of the specified type and share it with a group containing the writer
-func (c *ToznySDKV3) CreateSecret(ctx context.Context, secret CreateSecretOptions) (*Secret, error) {
+func (c *ToznySDKV3) CreateSecret(ctx context.Context, secretDetails CreateSecretOptions) (*Secret, error) {
 	// once ToznySDKV3 is updated, this will be c.RealmName if secret.RealmName is empty
-	realmName := secret.RealmName
-	err := ValidateSecret(secret)
+	var realmName string
+	if secretDetails.RealmName != "" {
+		realmName = secretDetails.RealmName
+	} else if c.CurrentIdentity.Realm != "" {
+		realmName = c.CurrentIdentity.Realm
+	} else {
+		return nil, fmt.Errorf("CreateSecret: No realm name was provided.")
+	}
+	err := ValidateSecret(secretDetails)
 	if err != nil {
 		return nil, err
 	}
+	ownerClientID, err := uuid.Parse(c.StorageClient.ClientID)
+	if err != nil {
+		return nil, err
```

Comment:
remember that a key reason for this change is to return an informative error message, suggest 
```golang

return nil, errors.New(fmt.Sprintf("CreateSecret: client id must be a valid uuid, got %s", .StorageClient.ClientID))
```
```

Reply:
got it! will update these error messages throughout.